Source/AsInterface.cs

```diff
@@ -83,7 +86,7 @@ public override bool CallBase
 			set { this.owner.CallBase = value; }
 		}
 
-		public override DefaultValue DefaultValue
+		public override Func<IDefaultValueProvider> DefaultValue
```

Comment:
I suppose the idea of using `Func` here might be lazy instantiation, in case that the provider is a "heavy" object. Given that the default value provider is probably used all the time, I'm not positively sure that lazy instantiation would have practical benefits, though.